nestkernel/mpi_manager.h

```diff
@@ -548,6 +556,16 @@ MPIManager::increase_buffer_size_spike_data()
   }
 }
 
+inline void
+MPIManager::decrease_buffer_size_spike_data()
+{
+  assert( adaptive_spike_buffers_ );
+  if ( buffer_size_spike_data_ / shrink_factor_buffer_spike_data_ > 4.0 * get_num_processes() )
```

Comment:
Could you add a comment to explain the lower limit, especially the factor four?

Reply:
`2.0 * get_num_processes()` represents the minimum allowable size (as initialised). I have doubled this to distinguish the minimum buffer size set through shrinkage (initially for debugging purposes), and by:

`kernel().mpi_manager.set_buffer_size_spike_data( 2 * kernel().mpi_manager.get_num_processes() );`

The lower limit through shrinkage can be reduced, but I suggest otherwise. It is unlikely that activity is so low that all ranks will communicate fewer than 4 spikes to any other rank (once buffers have grown beyond `4.0 * get_num_processes()`).

Reply:
@H27CK Could you add this in compact for as a comment in the code, so that we still understand the factor 4 when looking at this code three years from now?